server/commands/documentUpdater.js

```diff
@@ -0,0 +1,100 @@
+// @flow
+import { Document, User, Collection, Event } from "../models";
+import { sequelize } from "../sequelize";
+
+export async function documentUpdater(
+  document: Document,
+  user: User,
+  collection?: Collection,
+  {
+    id,
+    title,
+    text,
+    publish,
+    autosave,
+    done,
+    templateId,
+    append,
+    collectionId,
+    editorVersion,
+    parentDocumentId,
+  }: Object,
+  ip: String
+): Document {
+  // Update document
+  const previousTitle = document.title;
+  if (title) document.title = title;
+  if (editorVersion) document.editorVersion = editorVersion;
+  if (templateId) document.templateId = templateId;
+  document.text = append ? document.text + text : text;
+
+  document.lastModifiedById = user.id;
+
+  let transaction;
+  let updatedDocument;
+  try {
+    transaction = await sequelize.transaction();
+
+    if (publish) {
+      if (!document.publishedAt && !document.collection && collectionId) {
+        document.collectionId = collectionId;
```

Comment:
Could we do policies checks inside the commands? I checked the other commands, and they didn't have the checks. Is it by design?

Reply:
It's by design right now to do policy checks in controllers. It makes the commands more reusable – I'm happy to reconsider this more holisitically.